diff --git a/app/src/main/AndroidManifest.xml b/app/src/main/AndroidManifest.xml
index 4c47844..3c43e5e 100644
--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.example.x6.androidmachinetest">
-    <!--android:sharedUserId="android.uid.system">-->
+    package="com.example.x6.androidmachinetest"
+    android:sharedUserId="android.uid.system">
     <uses-permission android:name="android.permission.WRITE_SETTINGS" />
     <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
     <uses-permission android:name = "android.permission.DISABLE_KEYGUARD"/>
@@ -24,7 +24,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".Activity.DebugActivity"
+        <activity android:name=".Activity.StartActivity"
             android:configChanges="keyboardHidden|mcc|mnc|keyboard|orientation|touchscreen|screenLayout">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
@@ -32,7 +32,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
-        <activity android:name=".Activity.StartActivity"
+        <activity android:name=".Activity.DebugActivity"
             android:configChanges="keyboardHidden|mcc|mnc|keyboard|orientation|touchscreen|screenLayout"/>
         <activity android:name=".Activity.TestActivity"
             android:configChanges="keyboardHidden|mcc|mnc|keyboard|orientation|touchscreen|screenLayout"/>
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/StartActivity.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/StartActivity.java
index cb13019..44a9b20 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/StartActivity.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/StartActivity.java
@@ -145,10 +145,11 @@ public class StartActivity extends Activity {
             strings[5] = PreMachineInfo.K10;
             setTypeView(typeNum,strings);
         }else if (getMachineInfo.getCupName().equals(PreMachineInfo.H3)){
-            typeNum = 2;
+            typeNum = 3;
             String[] strings = new String[typeNum];
             strings[0] = PreMachineInfo.B301;
             strings[1] = PreMachineInfo.B301_LVDS;
+            strings[2] = PreMachineInfo.B301D_ELE01;
             setTypeView(typeNum,strings);
         }else if(getMachineInfo.getCupName().equals(PreMachineInfo.RK3288)){
             typeNum = 2;
@@ -173,7 +174,8 @@ public class StartActivity extends Activity {
                     if(machineInfoData.typeName.equals(PreMachineInfo.Y5)
                             ||machineInfoData.typeName.equals(PreMachineInfo.Y7)
                             ||machineInfoData.typeName.equals(PreMachineInfo.Y8)
-                            ||machineInfoData.typeName.equals(PreMachineInfo.Y10)){
+                            ||machineInfoData.typeName.equals(PreMachineInfo.Y10)
+                            ||machineInfoData.typeName.equals(PreMachineInfo.B301D_ELE01)){
                         machineInfoData.ISRS485 = true;
 
                         /*
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/TestActivity.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/TestActivity.java
index 428a03a..ecefe3b 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/TestActivity.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/TestActivity.java
@@ -85,6 +85,7 @@ public class TestActivity extends Activity {
         int temp = currentPosition - 1;
         String Pass = "测试通过";
         String NotPass = "测试不通过";
+        String NotSupport = "设备不支持";
         TestDataBaseUtils testDataBaseUtils = TestDataBaseUtils.getTestDataBaseUtils(TestActivity.this.getApplicationContext());
         if(speakTest!=null){
             if(temp == speakTest.getPosition()){
@@ -93,6 +94,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateSpk(1);
                 }else if(result.speakTestResult.equals(NotPass)){
                     testDataBaseUtils.updateSpk(0);
+                }else if(result.speakTestResult.equals(NotSupport)){
+                    testDataBaseUtils.updateSpk(-1);
                 }
             }
         }
@@ -103,6 +106,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateHp(1);
                 }else if(result.headPhoneTestResult.equals(NotPass)){
                     testDataBaseUtils.updateHp(0);
+                }else if(result.headPhoneTestResult.equals(NotSupport)){
+                    testDataBaseUtils.updateHp(-1);
                 }
             }
         }
@@ -113,6 +118,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateRecord(1);
                 }else if(result.recordTest.equals(NotPass)){
                     testDataBaseUtils.updateRecord(0);
+                }else if(result.recordTest.equals(NotSupport)){
+                    testDataBaseUtils.updateRecord(-1);
                 }
             }
         }
@@ -123,6 +130,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateEth(1);
                 }else if(result.ethTest.equals(NotPass)){
                     testDataBaseUtils.updateEth(0);
+                }else if(result.ethTest.equals(NotSupport)){
+                    testDataBaseUtils.updateEth(-1);
                 }
             }
         }
@@ -133,6 +142,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateWifi(1);
                 }else if(result.wifiTest.equals(NotPass)){
                     testDataBaseUtils.updateWifi(0);
+                }else if(result.wifiTest.equals(NotSupport)){
+                    testDataBaseUtils.updateWifi(-1);
                 }
             }
         }
@@ -143,6 +154,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateBT(1);
                 }else if(result.blueToothTest.equals(NotPass)){
                     testDataBaseUtils.updateBT(0);
+                }else if(result.blueToothTest.equals(NotSupport)){
+                    testDataBaseUtils.updateBT(-1);
                 }
             }
         }
@@ -153,6 +166,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateRS232(1);
                 }else if(result.RS232Test.equals(NotPass)){
                     testDataBaseUtils.updateRS232(0);
+                }else if(result.RS232Test.equals(NotSupport)){
+                    testDataBaseUtils.updateRS232(-1);
                 }
             }
         }
@@ -163,6 +178,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateRS485(1);
                 }else if(result.RS485Test.equals(NotPass)){
                     testDataBaseUtils.updateRS485(0);
+                }else if(result.RS485Test.equals(NotSupport)){
+                    testDataBaseUtils.updateRS485(-1);
                 }
             }
         }
@@ -173,6 +190,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateRS485(1);
                 }else if(result.RS485Test.equals(NotPass)){
                     testDataBaseUtils.updateRS485(0);
+                }else if(result.RS485Test.equals(NotSupport)){
+                    testDataBaseUtils.updateRS485(-1);
                 }
             }
         }
@@ -183,6 +202,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateUSB(1);
                 }else if(result.USBTest.equals(NotPass)){
                     testDataBaseUtils.updateUSB(0);
+                }else if(result.USBTest.equals(NotSupport)){
+                    testDataBaseUtils.updateUSB(-1);
                 }
             }
         }
@@ -193,6 +214,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateGPIO(1);
                 }else if(result.GpioTest.equals(NotPass)){
                     testDataBaseUtils.updateGPIO(0);
+                }else if(result.GpioTest.equals(NotSupport)){
+                    testDataBaseUtils.updateGPIO(-1);
                 }
             }
         }
@@ -203,6 +226,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateScreen(1);
                 }else if(result.ScreenTest.equals(NotPass)){
                     testDataBaseUtils.updateScreen(0);
+                }else if(result.ScreenTest.equals(NotSupport)){
+                    testDataBaseUtils.updateScreen(-1);
                 }
             }
         }
@@ -213,6 +238,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateTS(1);
                 }else if(result.TouchScreenTest.equals(NotPass)){
                     testDataBaseUtils.updateTS(0);
+                }else if(result.TouchScreenTest.equals(NotSupport)){
+                    testDataBaseUtils.updateTS(-1);
                 }
             }
         }
@@ -223,6 +250,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateMobileNet(1);
                 }else if(result.mobileNetTest.equals(NotPass)){
                     testDataBaseUtils.updateMobileNet(0);
+                }else if(result.mobileNetTest.equals(NotSupport)){
+                    testDataBaseUtils.updateMobileNet(-1);
                 }
             }
         }
@@ -233,6 +262,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateTF(1);
                 }else if(result.TFCARDTest.equals(NotPass)){
                     testDataBaseUtils.updateTF(0);
+                }else if(result.TFCARDTest.equals(NotSupport)){
+                    testDataBaseUtils.updateTF(-1);
                 }
             }
         }
@@ -243,6 +274,8 @@ public class TestActivity extends Activity {
                     testDataBaseUtils.updateGPS(1);
                 }else if(result.gpsTest.equals(NotPass)){
                     testDataBaseUtils.updateGPS(0);
+                }else if(result.gpsTest.equals(NotSupport)){
+                    testDataBaseUtils.updateGPS(-1);
                 }
             }
         }
@@ -264,13 +297,19 @@ public class TestActivity extends Activity {
                     bundle = msg.getData();
                     pos = bundle.getInt("pos");
                     testProject[pos].setTextColor(Color.GREEN);
-                    setResult(true);
+                    setTestResult(1);
                     break;
                 case -3:
                     bundle = msg.getData();
                     pos = bundle.getInt("pos");
                     testProject[pos].setTextColor(Color.RED);
-                    setResult(false);
+                    setTestResult(0);
+                    break;
+                case -1:
+                    bundle = msg.getData();
+                    pos = bundle.getInt("pos");
+                    testProject[pos].setTextColor(Color.argb(0,0xf1,0x9e,0x9e));
+                    setTestResult(-1);
                     break;
             }
         }
@@ -281,158 +320,275 @@ public class TestActivity extends Activity {
 
 
 
-    public static void syncResultList(Handler handler,int i,boolean isPass){
+    public static void syncResultList(Handler handler,int i,int isPass){
         Bundle bundle = new Bundle();
         Message message = new Message();
         bundle.putInt("pos",i);
-        if(isPass)
-            message.what = 3;
-        else
-            message.what = -3;
+        switch (isPass){
+            case 1:
+                message.what = 3;
+                break;
+            case -1:
+                message.what = -1;
+                break;
+            case 0:
+                message.what = -3;
+                break;
+        }
         message.setData(bundle);
         handler.sendMessage(message);
     }
-    private void setResult(boolean isPass){
+    private void setTestResult(int isPass){
         if(speakTest != null){
             if(currentPosition == speakTest.getPosition()) {
-                if (isPass)
-                    result.speakTestResult = "测试通过";
-                else
-                    result.speakTestResult = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.speakTestResult = "测试通过";
+                        break;
+                    case 0:
+                        result.speakTestResult = "测试不通过";
+                        break;
+                    case -1:
+                        result.speakTestResult = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(headhhoneTest != null){
             if(currentPosition == headhhoneTest.getPosition()){
-                if(isPass)
-                    result.headPhoneTestResult = "测试通过";
-                else
-                    result.headPhoneTestResult = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.headPhoneTestResult = "测试通过";
+                        break;
+                    case 0:
+                        result.headPhoneTestResult = "测试不通过";
+                        break;
+                    case -1:
+                        result.headPhoneTestResult = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(recordTest != null){
-            if(currentPosition == recordTest.getPosition()){
-                if(isPass)
+            switch (isPass){
+                case 1:
                     result.recordTest = "测试通过";
-                else
+                    break;
+                case 0:
                     result.recordTest = "测试不通过";
+                    break;
+                case -1:
+                    result.recordTest = "设备不支持";
+                    break;
             }
         }
 
         if(ethTest != null){
             if(currentPosition == ethTest.getPosition()){
-                if(isPass)
-                    result.ethTest = "测试通过";
-                else
-                    result.ethTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.ethTest = "测试通过";
+                        break;
+                    case 0:
+                        result.ethTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.ethTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(wifiTest != null){
             if(currentPosition == wifiTest.getPosition()){
-                if(isPass)
-                    result.wifiTest = "测试通过";
-                else
-                    result.wifiTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.wifiTest = "测试通过";
+                        break;
+                    case 0:
+                        result.wifiTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.wifiTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(blueToothTest != null){
             if(currentPosition == blueToothTest.getPosition()){
-                if(isPass)
-                    result.blueToothTest = "测试通过";
-                else
-                    result.blueToothTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.blueToothTest = "测试通过";
+                        break;
+                    case 0:
+                        result.blueToothTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.blueToothTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(serial232Test != null){
             if(currentPosition == serial232Test.getPosition()){
-                if(isPass)
-                    result.RS232Test = "测试通过";
-                else
-                    result.RS232Test = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.RS232Test = "测试通过";
+                        break;
+                    case 0:
+                        result.RS232Test = "测试不通过";
+                        break;
+                    case -1:
+                        result.RS232Test = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(serial485Odd != null){
             if(currentPosition == serial485Odd.getPosition()){
-                if(isPass)
-                    result.RS485Test = "测试通过";
-                else
-                    result.RS485Test = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.RS485Test = "测试通过";
+                        break;
+                    case 0:
+                        result.RS485Test = "测试不通过";
+                        break;
+                    case -1:
+                        result.RS485Test = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(serial485Even != null){
             if(currentPosition == serial485Even.getPosition()){
-                if(isPass)
-                    result.RS485Test = "测试通过";
-                else
-                    result.RS485Test = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.RS485Test = "测试通过";
+                        break;
+                    case 0:
+                        result.RS485Test = "测试不通过";
+                        break;
+                    case -1:
+                        result.RS485Test = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(usbTest != null){
             if(currentPosition == usbTest.getPosition()){
-                if(isPass)
-                    result.USBTest = "测试通过";
-                else
-                    result.USBTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.USBTest = "测试通过";
+                        break;
+                    case 0:
+                        result.USBTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.USBTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(tfcardTest != null){
             if(currentPosition == tfcardTest.getPosition()){
-                if(isPass)
-                    result.TFCARDTest = "测试通过";
-                else
-                    result.TFCARDTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.TFCARDTest = "测试通过";
+                        break;
+                    case 0:
+                        result.TFCARDTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.TFCARDTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(gpioTest != null){
             if(currentPosition == gpioTest.getPosition()){
-                if(isPass)
-                    result.GpioTest = "测试通过";
-                else
-                    result.GpioTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.GpioTest = "测试通过";
+                        break;
+                    case 0:
+                        result.GpioTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.GpioTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(screenTets != null){
             if(currentPosition == screenTets.getPosition()){
-                if(isPass)
-                    result.ScreenTest = "测试通过";
-                else
-                    result.ScreenTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.ScreenTest = "测试通过";
+                        break;
+                    case 0:
+                        result.ScreenTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.ScreenTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(touchTest != null){
             if(currentPosition == touchTest.getPosition()){
-                if(isPass)
-                    result.TouchScreenTest = "测试通过";
-                else
-                    result.TouchScreenTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.TouchScreenTest = "测试通过";
+                        break;
+                    case 0:
+                        result.TouchScreenTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.TouchScreenTest = "设备不支持";
+                        break;
+                }
             }
         }
 
         if(mobileNet != null){
             if(currentPosition == mobileNet.getPosition()){
-                if(isPass)
-                    result.mobileNetTest = "测试通过";
-                else
-                    result.mobileNetTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.mobileNetTest = "测试通过";
+                        break;
+                    case 0:
+                        result.mobileNetTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.mobileNetTest = "设备不支持";
+                        break;
+                }
             }
         }
         if(gpsTest != null){
             if(currentPosition == gpsTest.getPosition()){
-                if(isPass)
-                    result.gpsTest = "测试通过";
-                else
-                    result.gpsTest = "测试不通过";
+                switch (isPass){
+                    case 1:
+                        result.gpsTest = "测试通过";
+                        break;
+                    case 0:
+                        result.gpsTest = "测试不通过";
+                        break;
+                    case -1:
+                        result.gpsTest = "设备不支持";
+                        break;
+                }
             }
         }
 
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/BlueTooth.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/BlueTooth.java
index 9ad2fbf..d668338 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/BlueTooth.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/BlueTooth.java
@@ -49,7 +49,7 @@ public class BlueTooth extends AutoCommon {
                 }
                 while (!mBluetoothAdapter.isEnabled()){
                     if(System.currentTimeMillis() - startTime > 5*1000){
-                        TestActivity.syncResultList(handler,getPosition(),false); //蓝牙打开超时
+                        TestActivity.syncResultList(handler,getPosition(),0); //蓝牙打开超时
                         handler.post(new Runnable() {
                             @Override
                             public void run() {
@@ -68,7 +68,7 @@ public class BlueTooth extends AutoCommon {
                     }
                 }
                 if(mBluetoothAdapter.isEnabled()){
-                    TestActivity.syncResultList(handler,getPosition(),true);
+                    TestActivity.syncResultList(handler,getPosition(),1);
                     handler.post(new Runnable() {
                         @Override
                         public void run() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Eth.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Eth.java
index 4c4e710..3c90de2 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Eth.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Eth.java
@@ -170,9 +170,9 @@ public class Eth extends AutoCommon {
                 getTextView().setText(message);
                 getTextView().setTextColor(color);
                 if(color == Color.RED)
-                    TestActivity.syncResultList(handler,getPosition(),false);
+                    TestActivity.syncResultList(handler,getPosition(),1);
                 else if(color == Color.BLUE)
-                    TestActivity.syncResultList(handler,getPosition(),true);
+                    TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
     }
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPIO.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPIO.java
index 93f7fc5..7470434 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPIO.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPIO.java
@@ -74,7 +74,7 @@ public class GPIO extends AutoCommon{
         }else{
             getTextView().setText("当前没有找到GPIO， GPIO测试失败");
             handler.sendEmptyMessage(0);
-            TestActivity.syncResultList(handler,getPosition(),false);
+            TestActivity.syncResultList(handler,getPosition(),0);
             return false;
         }
     }
@@ -126,7 +126,11 @@ public class GPIO extends AutoCommon{
         }
 
         handler.sendEmptyMessage(0);
-        TestActivity.syncResultList(handler,getPosition(),isOk);
+        if(isOk){
+            TestActivity.syncResultList(handler,getPosition(),1);
+        }else{
+            TestActivity.syncResultList(handler,getPosition(),0);
+        }
         getReTest().setEnabled(true);
     }
 
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPS.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPS.java
index 6ebf211..a1e98ac 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPS.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/GPS.java
@@ -65,7 +65,7 @@ public class GPS extends Common {
                 getNo().setTextColor(Color.WHITE);
                 notSupport.setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler, getPosition(), true);
+                TestActivity.syncResultList(handler, getPosition(), 1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -77,7 +77,7 @@ public class GPS extends Common {
                 getYes().setTextColor(Color.WHITE);
                 notSupport.setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler, getPosition(), false);
+                TestActivity.syncResultList(handler, getPosition(), 0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
@@ -103,7 +103,7 @@ public class GPS extends Common {
                 getNo().setTextColor(Color.WHITE);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler, getPosition(), true);
+                TestActivity.syncResultList(handler, getPosition(), -1);
             }
         });
 
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Headhhone.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Headhhone.java
index 5e46543..05a0e94 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Headhhone.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Headhhone.java
@@ -34,7 +34,7 @@ public class Headhhone extends Common{
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -44,7 +44,7 @@ public class Headhhone extends Common{
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/MobileNet.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/MobileNet.java
index 00aa683..f684fd1 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/MobileNet.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/MobileNet.java
@@ -40,7 +40,7 @@ public class MobileNet extends Common {
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -50,7 +50,7 @@ public class MobileNet extends Common {
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Record.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Record.java
index 2149e0c..fc9dda9 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Record.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Record.java
@@ -31,7 +31,7 @@ public class Record extends Common {
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -41,7 +41,7 @@ public class Record extends Common {
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Screen.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Screen.java
index 0cb1820..b3f92d5 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Screen.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Screen.java
@@ -29,7 +29,7 @@ public class Screen extends Common {
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getYes().setOnClickListener(new View.OnClickListener() {
@@ -39,7 +39,7 @@ public class Screen extends Common {
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial232Control.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial232Control.java
index 42e0c96..20853d7 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial232Control.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial232Control.java
@@ -207,7 +207,11 @@ public class Serial232Control extends Serial232 {
                         isOk = false;
                     }
                 }
-                TestActivity.syncResultList(handler,getPosition(),isOk);
+                if(isOk){
+                    TestActivity.syncResultList(handler,getPosition(),1);
+                }else{
+                    TestActivity.syncResultList(handler,getPosition(),0);
+                }
                 handler.sendEmptyMessage(0);
                 handler.post(new Runnable() {
                     @Override
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Even.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Even.java
index 08eff25..bdcd222 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Even.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Even.java
@@ -129,7 +129,11 @@ public class Serial485Even extends AutoCommon{
                     serial485Remark("串口3 -> 串口2 失败");
                     isSerial485Ok = false;
                 }
-                TestActivity.syncResultList(handler,getPosition(),isSerial485Ok);
+                if(isSerial485Ok){
+                    TestActivity.syncResultList(handler,getPosition(),1);
+                }else{
+                    TestActivity.syncResultList(handler,getPosition(),0);
+                }
                 handler.sendEmptyMessage(0);
                 handler.post(new Runnable() {
                     @Override
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Odd.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Odd.java
index c11b607..3e4d76a 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Odd.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Serial485Odd.java
@@ -41,7 +41,7 @@ public class Serial485Odd extends Common{
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -51,7 +51,7 @@ public class Serial485Odd extends Common{
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
 
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Speak.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Speak.java
index d5e3ab3..9cc7cc2 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Speak.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Speak.java
@@ -41,7 +41,7 @@ public class Speak extends Common{
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -51,7 +51,7 @@ public class Speak extends Common{
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/TFCARD.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/TFCARD.java
index 73df396..c245c80 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/TFCARD.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/TFCARD.java
@@ -33,7 +33,7 @@ public class TFCARD extends Common {
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -43,7 +43,7 @@ public class TFCARD extends Common {
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Touch.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Touch.java
index 274f074..292df41 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Touch.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Touch.java
@@ -30,7 +30,7 @@ public class Touch extends Common {
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -40,7 +40,7 @@ public class Touch extends Common {
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/USB.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/USB.java
index 08272c5..ad244e4 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/USB.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/USB.java
@@ -34,7 +34,7 @@ public class USB extends Common{
                 getNo().setEnabled(false);
                 getNo().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),true);
+                TestActivity.syncResultList(handler,getPosition(),1);
             }
         });
         getNo().setOnClickListener(new View.OnClickListener() {
@@ -44,7 +44,7 @@ public class USB extends Common{
                 getNo().setEnabled(false);
                 getYes().setTextColor(Color.WHITE);
                 handler.sendEmptyMessage(0);
-                TestActivity.syncResultList(handler,getPosition(),false);
+                TestActivity.syncResultList(handler,getPosition(),0);
             }
         });
         getReTest().setOnClickListener(new View.OnClickListener() {
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Wifi.java b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Wifi.java
index ccbba91..6c09526 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Wifi.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Activity/UI/Wifi.java
@@ -33,10 +33,10 @@ public class Wifi extends AutoCommon {
                 case 1:
                     appendWifiState(wifiControl.getMsg(msg),Color.BLUE);
                     handlerP.sendEmptyMessage(0);
-                    TestActivity.syncResultList(handlerP,getPosition(),true);
+                    TestActivity.syncResultList(handlerP,getPosition(),1);
                     break;
                 case 2:
-                    TestActivity.syncResultList(handlerP,getPosition(),false); //wifi超过5S打不开，那就识别为WIFI有故障
+                    TestActivity.syncResultList(handlerP,getPosition(),0); //wifi超过5S打不开，那就识别为WIFI有故障
                     handlerP.sendEmptyMessage(0);
                     getTextView().setText("WIFI 打开超时，请检查WIFI是否有故障");
                     getTextView().setTextColor(Color.RED);
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Core/MachineInfoData.java b/app/src/main/java/com/example/x6/androidmachinetest/Core/MachineInfoData.java
index 6c19382..468509a 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Core/MachineInfoData.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Core/MachineInfoData.java
@@ -58,6 +58,7 @@ public class MachineInfoData {
     public boolean ISPHONE = false;
 
     public boolean SCREEN = false;
+    public boolean CAMERA_ONE = false;
     public boolean TOUCH_SCREEN = false;
     public int GPIONUM = -1;
     public int[] GPIO = null;
@@ -92,6 +93,7 @@ public class MachineInfoData {
         ISWIFI = false;
         ISPHONE = false;
         SCREEN = false;
+        CAMERA_ONE = false;
         TOUCH_SCREEN = false;
         GPIONUM = -1;
         GPIO = null;
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Core/PlanTest.java b/app/src/main/java/com/example/x6/androidmachinetest/Core/PlanTest.java
index 36b24a0..1c8608e 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Core/PlanTest.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Core/PlanTest.java
@@ -29,6 +29,11 @@ public class PlanTest {
     private GetMachineInfo getMachineInfo = new GetMachineInfo();
 
 
+    /*************************************************************************************************************/
+    /*      initMachineInfoData() 函数用于初始化数据，把要测试的项目写到一个集合里面，添加测试项目这里一定要加   */
+    /*      setPlanStep() 函数获取添加好的集合，对测试项目进行一个计划排版，添加测试项目，这里也一定要改动       */
+    /*************************************************************************************************************/
+
     public void initMachineInfoData(Context context,String typeName,String UUID,Boolean isRS485,Boolean isWifiTest,Boolean isPHONE){
         machineInfoData.cpuName = getMachineInfo.getCupName();
         machineInfoData.typeName = typeName;
@@ -105,7 +110,8 @@ public class PlanTest {
             machineInfoData.GPIO = gpio;
             machineInfoData.RTC = true;
         }
-        if(machineInfoData.typeName.equals(PreMachineInfo.B301) || machineInfoData.typeName.equals(PreMachineInfo.B301_LVDS)){
+        if(machineInfoData.typeName.equals(PreMachineInfo.B301) || machineInfoData.typeName.equals(PreMachineInfo.B301_LVDS)
+                || machineInfoData.typeName.equals(PreMachineInfo.B301D_ELE01)){
             machineInfoData.RS232_1 = PreMachineInfo.ttyS0;
             machineInfoData.RS232_2 = PreMachineInfo.ttyS1;
             if(machineInfoData.ISRS485) {
@@ -125,6 +131,10 @@ public class PlanTest {
                 machineInfoData.SCREEN = true;
                 machineInfoData.TOUCH_SCREEN = true;
             }
+            if(machineInfoData.typeName.equals(PreMachineInfo.B301D_ELE01)){
+                machineInfoData.SCREEN = true;
+                machineInfoData.CAMERA_ONE = true;
+            }
             machineInfoData.RTC = true;
         }
         if(machineInfoData.typeName.equals(PreMachineInfo.B701)||machineInfoData.typeName.equals(PreMachineInfo.B601)){
@@ -152,8 +162,9 @@ public class PlanTest {
             /*
             这里如果需要测试屏幕和触摸就把注释打开
              */
-            machineInfoData.SCREEN = true;
-            machineInfoData.TOUCH_SCREEN = true;
+//            machineInfoData.SCREEN = true;
+//            machineInfoData.TOUCH_SCREEN = true;
+
             machineInfoData.GPIONUM = 4;
             int[] gpio = new int[machineInfoData.GPIONUM];
             gpio[0] = 32 +26;//输入
@@ -173,6 +184,8 @@ public class PlanTest {
             vector.add(PreMachineInfo.SPEAK_TEST);
         if(machineInfoData.HEADPHONE)
             vector.add(PreMachineInfo.HEADPHONE_TEST);
+        if(machineInfoData.CAMERA_ONE)
+            vector.add(PreMachineInfo.CAMERA_ONE_TEST);
         if(machineInfoData.RECORD)
             vector.add(PreMachineInfo.RECORD_TEST);
         if(machineInfoData.wifiModule != null ) {
@@ -280,6 +293,7 @@ public class PlanTest {
         machineInfoData.RECORD = false;
         machineInfoData.SPEAK = false;
         machineInfoData.HEADPHONE = false;
+        machineInfoData.CAMERA_ONE = false;
         machineInfoData.TFCARD = false;
         machineInfoData.USB = false;
         machineInfoData.ISWIFI = false;
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/Core/PreMachineInfo.java b/app/src/main/java/com/example/x6/androidmachinetest/Core/PreMachineInfo.java
index 69a8416..7d57d57 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/Core/PreMachineInfo.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/Core/PreMachineInfo.java
@@ -24,6 +24,8 @@ public class PreMachineInfo {
     public final static String K10 = "K10";
     public final static String B301 = "B301";
     public final static String B301_LVDS = "B301_LVDS";
+    public final static String B301D_ELE01 = "B301D_ELE01";
+
     public final static String B701 = "B701";
     public final static String B601 = "B601";
 
@@ -55,6 +57,7 @@ public class PreMachineInfo {
     public final static String SPEAK_TEST = "外放";
     public final static String HEADPHONE_TEST = "耳机";
     public final static String RECORD_TEST = "录音";
+    public final static String CAMERA_ONE_TEST = "单个摄像头录像";
     public final static String WFIF_TEST = "WiFi";
     public final static String BLUETOOTH_TEST = "蓝牙";
     public final static String MOBILE_NET_TEST = "移动网络";
diff --git a/app/src/main/java/com/example/x6/androidmachinetest/DataBase/TestDataBaseUtils.java b/app/src/main/java/com/example/x6/androidmachinetest/DataBase/TestDataBaseUtils.java
index 37c272c..b71c969 100644
--- a/app/src/main/java/com/example/x6/androidmachinetest/DataBase/TestDataBaseUtils.java
+++ b/app/src/main/java/com/example/x6/androidmachinetest/DataBase/TestDataBaseUtils.java
@@ -354,7 +354,9 @@ public class TestDataBaseUtils {
             PassTimesValue += 1;
             TestTimesValue += 1;
         }else if(ISPass == -1){
-
+            PassTimesValue = 0;
+            TestTimesValue = 0;
+            NotPassTimesValue = 0;
         }else{
             return false;
         }
diff --git a/app/src/main/res/layout/activity_auto_common.xml b/app/src/main/res/layout/activity_auto_common.xml
index e3d5a1b..305cb4d 100644
--- a/app/src/main/res/layout/activity_auto_common.xml
+++ b/app/src/main/res/layout/activity_auto_common.xml
@@ -5,5 +5,7 @@
 
     <LinearLayout
         android:layout_width="match_parent"
-        android:layout_height="match_parent"></LinearLayout>
+        android:layout_height="match_parent">
+    </LinearLayout>
+
 </ScrollView>
\ No newline at end of file
